nemo_text_processing/text_normalization/hy/taggers/cardinal.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pynini
+from pynini.lib import pynutil
+
+from nemo_text_processing.text_normalization.en.graph_utils import NEMO_DIGIT, GraphFst, delete_space, insert_space
+from nemo_text_processing.text_normalization.hy.utils import get_abs_path
+
+
+class CardinalFst(GraphFst):
+    """
+    Finite state transducer for classifying cardinals, e.g.
+        55 -> cardinal { integer: "հիսունհինգ" }
+    """
+
+    def __init__(self):
+        super().__init__(name="cardinal", kind="classify")
+
+        zero = pynini.string_map([("0", "զրո")])
+        digits = pynini.string_file(get_abs_path("data/numbers/digit.tsv")).invert()
+        digits_no_one = (NEMO_DIGIT - "1") @ digits
+
+        ties = pynini.string_file(get_abs_path("data/numbers/ties.tsv")).invert()
+        ties_unit = digits
+        double_digits = (pynini.cross("1", "տասը") | ties) + pynutil.delete("0") | (
+            pynini.cross("1", "տասն") | ties
+        ) + ties_unit
+
+        self.all_double_digits = double_digits.optimize()
+
+        one_to_all_tens = digits | double_digits
+        self.one_to_all_tens = one_to_all_tens.optimize()
+
+        hundreds_parts = (pynutil.delete("0") + insert_space + digits) | (insert_space + double_digits)
+        one_hundreds = pynini.cross("1", "հարյուր") + (pynutil.delete("00") | hundreds_parts)
```

Comment:
If you ran 

1) `normalize.py` on just --text="**123**" you will get the same "**123**" instead of normalized "**հարյուր քսաներեք**" 
2) ՝python normalize.py --text="123թվ․ Տրդատը կառուցեց 3 ամրոց"՝ will again output the same text

Reply:
In the first case you mentioned, everything works fine on my side. As for the second case there is no 'թվ.' only 'թ.' and 'թթ.' and don't use Armenian dot it does not work (use English). By the way thank you for this comment as measurements and measurement dates only worked with space after cardinal/decimal, now space is optional.

Reply:
Thanks for your response!

1) I cloned your fork and installed the project with the `./reinstall.sh` . With CLI `python normalize.py --text="123" --
language=hy` I get output "**123**". With that being said, maybe you can guide on how to reproduce your project state ? as at the moment I get the afore-mentioned results for all the comments I mentioned.

2) How likely is to get an Armenian text with English dots ? Wouldn't it be useful to handle the punctuations of the target language?